spark/src/test/scala/org/apache/spark/sql/delta/optimize/DeltaReorgSuite.scala

```diff
@@ -133,6 +134,25 @@ class DeltaReorgSuite extends QueryTest
     }
   }
 
+  test("REORG TABLE UPGRADE UNIFORM from v1 to v2") {
+    val targetDf = spark.range(0, 100, 1, numPartitions = 5).toDF()
+    withTempDeltaTable(targetDf, enableDVs = false, conf = Seq(
+      DeltaConfigs.UNIVERSAL_FORMAT_ENABLED_FORMATS.defaultTablePropertyKey -> "iceberg",
+      DeltaConfigs.ICEBERG_COMPAT_V1_ENABLED.defaultTablePropertyKey -> "true"
```

Comment:
for `IcebergCompatV1`, we also need column mapping to be enabled - or we can also choose to remove all configurations here and directly upgrade a vanilla delta table to `IcebergCompatV2`.